backends/imgui_impl_dx9.cpp

```diff
@@ -46,6 +46,11 @@ struct ImGui_ImplDX9_Data
     int                         VertexBufferSize;
     int                         IndexBufferSize;
 
+    IDirect3DVertexDeclaration9* pInputLayout;
```

Comment:
Suggesting to add a comment at the top of those 4 fields, e.g. `// Support for Shader Pipeline mode (rare)`.
Move the `bool` at the top of this 4 variables block.

Reply:
I have moved the variable position and added comments. It may be surprising that the number of devices supporting shader model 2 is very large, so I didn't use (rare), this should be acceptable.